src/adapter/src/coord/appends.rs

```diff
@@ -288,6 +288,21 @@ impl Coordinator {
             timestamp,
             advance_to,
         } = self.get_local_write_ts().await;
+
+        // While we're flipping on the feature flags for persist-txn tables and
+        // the separated Postgres timestamp oracle, we also need to confirm
+        // leadership on writes _after_ getting the timestamp and _before_
+        // writing anything to table shards. See the big comment on `init_txns`
+        // in the Storage controller for details.
+        //
+        // TODO: Remove this after both (either?) of the above features are on
+        // for good and no possibility of running the old code.
+        let () = self
+            .catalog
+            .confirm_leadership()
```

Comment:
Note to self: we'll need to run nightlies to understand the perf impact of this before merging

Reply:
Nightlies showed pretty much exactly the performance regression I'd expect for INSERTs/UPDATEs.

Otoh, I'm feeling a touch silly about slowing things down here before we're actually getting anything from persist-txn. I think what I'll do is comment this out for now with a `TODO(txn)` and reenable it in the same PR that turns persist-txn on in CI (which will have a second slowdown in INSERTs). That way we bundle up all the perf differences in one PR for easy evaluation